### DISCUSSION TITLE
Servo (no relays) operating generator choke.

### DISCUSSION THREAD

**Discussion by rederikus** [arendst/Tasmota#19176] [Support request]
I am using a RDS3218 servo which has Pulse Width 500 2500   uSec.  It moves 270 degrees.  However I am having trouble making it move more than about 30 degrees.  I am using the following parameters to configure it.

GPIO3  // Virtual relay for shutter open for PWM servo
GPIO4  // PWM servo motor for choke 
GPIO5  // Virtual relay for shutter close for PWM servo

backlog setOption80 1;shutterrelay1 1;Shuttermode 5;PWMfrequency 333;SetOption15 0
backlog ShutterPwmRange1 0.1, 0.5;ShutterOpenDuration1 0.75;ShutterCloseDuration1 0.75;ShutterMotorDelay1 0.2;SetOption 15 0;Restart 1

I have been at this for two days now without success.  I am only runnning to make it move through 180 degrees and start and stop it at pre-determined points.  This is for the choke lever on my generator.  I want no choke, half choke and full choke.  

PWMfrequency 333 is correct for this motor according to the manufacturer.  I believe that I have correctly set ShutterPwmRange usng.
Servo RDS3218 Pulse Width 500    2500   uSec
      divided by 5        100     500   uSec
      convert to mSec /1K   0.1    0.5  mSec
	  
I have made no other changes and am running and ESP32 DevKit 1 connecting only via WebUI with:
Program Version	13.0.0(tasmota32)
Build Date & Time	2023-06-26T10:54:38
Core/SDK Version	2_0_10/4.4.5.230614

Thanks in advance
Pete

**Comment by Jason2866** [comment]
I would try without shutter at all. Defining three buttons. Every Button fires a rule where you set the needed PWM value. Used together with `SetOption15 0`
